src/tainting/Dataflow_tainting.ml

```diff
@@ -1208,14 +1208,38 @@ let check_function_signature env fun_exp args args_taints =
   match (!hook_function_taint_signature, fun_exp) with
   | Some hook, { e = Fetch f; eorig = SameAs eorig } ->
       let* fparams, fun_sig = hook env.config eorig in
+      (* This function simply produces the corresponding taints to the
+          given argument, within the body of the function.
+      *)
+      (* Our first pass will be to substitute the args for taints.
+         We can't do this indiscriminately at the beginning, because
+         we might need to use some of the information of the pre-substitution
+         taints and the post-substitution taints, for instance the tokens.
+
+         So we will isolate this as a specific step to be applied as necessary.
+      *)
+      let first_pass_substitute_arg_with_taints arg =
+        match taints_of_sig_arg env fparams fun_exp args args_taints arg with
+        | None -> []
```

Comment:
I think it's better to keep `Taint_set.t` as long as it is possible. (Probably need to rethink that data structure too but that's another issue.)